src/seedu/addressbook/commands/Command.java

```diff
@@ -23,6 +23,8 @@ public Command(int targetIndex) {
         this.setTargetIndex(targetIndex);
     }
 
+    public abstract boolean isMutating();
```

Comment:
It is better to mention it returns true if the Command *potentially* mutates the data (header comments specify a contract, so try to be as precise as possible). E.g. Adding a duplicate person will not mutate data but the corresponding AddCommand object will still return true for this method.